src/pages/snippets/preloading-strategy.mdx

```diff
@@ -0,0 +1,197 @@
+---
+title: Preloading strategy
+description: "How to implement a module preloading strategy"
+tags: ["angular", "modules", "preloading", "router"]
+pubDate: Mar 8, 2023
+contributedBy: "@rubenperegrina"
+---
+
+If we have our application separated by modules and we apply lazy loading.
+Angular by default will load the modules as the user needs them.
+There are several ways to control the loading of modules.
+
+### Lazy loading
+
+Modules will be loaded as the user requires them.
+
+```typescript
+const routes: Routes = [
+  {
+    path: "contact",
+    loadChildren: () =>
+      import("./contact/contact.module").then((m) => m.ContactModule),
+  },
+  {
+    path: "about",
+    loadChildren: () =>
+      import("./about/about.module").then((m) => m.AboutModule),
+  },
+];
+
+@NgModule({
+  imports: [RouterModule.forRoot(routes)],
+  exports: [RouterModule],
+})
+export class AppRoutingModule {}
+
+//StandAlone API version
+
+//Lazy loading another routing config
+export const routes: Routes = [
+  {
+    path: "contact",
+    loadChildren: () =>
+      import("./contact/contact.routes").then((m) => m.CONTACT_ROUTES),
+  },
+  {
+    path: "about",
+    loadChildren: () =>
+      import("./about/about.routes").then((m) => m.ABOUT_ROUTES),
+  },
+];
+
+//Directly lazy loading a standalone component
+export const routes: Routes = [
+  {
+    path: "contact",
+    loadComponent: () =>
+      import("./contact/contact.component").then((m) => m.ContactComponent),
+  },
+  {
+    path: "about",
+    loadComponent: () =>
+      import("./about/about.component").then((m) => m.AboutComponent),
+  },
+];
+
+//In your main.ts add:
+bootstrapApplication(AppComponent, {
+  providers: [
+    importProvidersFrom(RouterModule.forRoot(routes))
+  ]
+})
+
+```
+
+### Preload all the modules
+
+In this way, all the modules of our application will be loaded at once and ready to use.
+
+```typescript
+@NgModule({
+  imports: [
+    RouterModule.forRoot(routes, {
+      preloadingStrategy: PreloadAllModules,
+    }),
+  ],
+  exports: [RouterModule],
+})
+export class AppRoutingModule {}
+
+//StandAlone API version
+//In your main.ts add:
+bootstrapApplication(AppComponent, {
+  providers: [
+    importProvidersFrom(RouterModule.forRoot(routes, {
```

Comment:
with new API you can do

` providers: [
    provideRouter(
      routes,
      withPreloading(PreloadAllModules)
    ),`